src/MonoTorrent.Client/MonoTorrent/TorrentCreator.cs

```diff
@@ -232,6 +234,11 @@ internal async Task<BEncodedDictionary> CreateAsync (string name, List<InputFile
             info["name"] = (BEncodedString) name;
             AddCommonStuff (torrent);
 
+            foreach (var file in files.Take (files.Count - 1)) {
+                file.Padding = (UsePadding && file.Length > 0) ? PieceLength - (file.Length % PieceLength) : 0;
```

Comment:
I just realised that this would add `PieceLength ` padding if `file.Length` is an exact multiple of `PieceLength`.

I added a test and fixed the bug!